### PR TITLE
Add option to bypass Edge installation for E2E tests (Linux)

### PR DESCRIPTION
**Pull request by SlavyMihov**
This change is needed for the support of Azure EFLoW (Edge For Linux on Windows).
The Mariner (Linux) VM that composes EFLoW comes with IoT Edge pre-provisioned.
This change allows E2E tests to omit installation of the _iotedge_ daemon and to use daemon pre-provisioned in the Mariner image.
This is an equivalent change for Linux to [#1188](https://github.com/Azure/iotedge/pull/1188).

### CODE REVIEW

File: scripts/linux/runE2ETest.sh [hunk 1]
```diff
@@ -1084,6 +1096,7 @@ function usage() {
     echo ' -testInfo                                      Contains comma delimiter test information, e.g. build number and id, source branches of build, edgelet and images.' 
     echo ' -testStartDelay                                Tests start after delay for applicable modules'
     echo ' -runtimeLogLevel                               Value of RuntimeLogLevel envivronment variable for EdgeAgent in Long Haul and Stress tests [Default: debug] (EdgeHub RuntimeLogLevel is set implicitly set to be the same with edgeAgent)'
+    echo ' -bypassEdgeInstallation                        Skip instaling iotedge (if already preinstalled)'
```
Comment (philipktlin) [inline scripts/linux/runE2ETest.sh:1099]:
```suggestion
    echo ' -bypassEdgeInstallation                        Skip installing iotedge (if already preinstalled)'
```

File: smoke/IotEdgeQuickstart/details/IotedgedLinux.cs [hunk 2]
```diff
@@ -147,31 +149,40 @@ public async Task VerifyModuleIsRunning(string name)
 
         public Task Install()
         {
-            const string PackageName = "iotedge";
+            if (false && this.requireEdgeInstallation)
```
Comment (philipktlin) [inline smoke/IotEdgeQuickstart/details/IotedgedLinux.cs:152]:
```suggestion
            if (this.requireEdgeInstallation)
```

File: smoke/IotEdgeQuickstart/details/IotedgedLinux.cs [hunk 3]
```diff
@@ -147,31 +149,40 @@ public async Task VerifyModuleIsRunning(string name)
...
-                commandName = "dpkg";
-                commandArgs = $"--force-confnew -i {this.archivePath}";
-            }
+                Console.WriteLine("Skipping iotedge installation.");
 
-            return Process.RunAsync(
-                commandName,
-                commandArgs,
-                300); // 5 min timeout because install can be slow on raspberry pi
+                return Task.FromResult(0);
```
Comment (philipktlin) [inline smoke/IotEdgeQuickstart/details/IotedgedLinux.cs:184]:
```suggestion
                return Task.CompletedTask;
```

File: scripts/linux/runE2ETest.sh (hunk 1 above)
Comment (SlavyMihov) [reply scripts/linux/runE2ETest.sh]: Thanks.

File: smoke/IotEdgeQuickstart/details/IotedgedLinux.cs (hunk 2 above)
Comment (SlavyMihov) [reply smoke/IotEdgeQuickstart/details/IotedgedLinux.cs]: My bad, fixing it.

File: smoke/IotEdgeQuickstart/details/IotedgedLinux.cs (hunk 3 above)
Comment (SlavyMihov) [reply smoke/IotEdgeQuickstart/details/IotedgedLinux.cs]: Thanks.

**Comment by philipktlin** [comment]
are we going to create a new platform in E2E tests release pipeline for Mariner (Linux) VM (that composes EFLoW comes with IoT Edge pre-provisioned) and every run will provision a new Mariner VM to run E2E tests for this new platform, is it right?

**Comment by SlavyMihov** [comment]
Yes, I am currently working on an [EFLoW stage](https://msazure.visualstudio.com/One/_releaseDefinition?definitionId=2189&_a=definition-tasks&environmentId=139080) in the Azure-IoT-Edge-E2E pipeline.
The Mariner VM will be provisioned each time the pipeline runs.
It is still work in progress.

**Pull request merged**